src/main/java/au/id/micolous/metrodroid/transit/kmt/KMTTrip.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * KMTTrip.java
+ *
+ * This file is part of FareBot.
+ * Learn more at: https://codebutler.github.io/farebot/
+ *
+ * Copyright (C) 2014-2016 Eric Butler <eric@codebutler.com>
+ * Copyright (C) 2016 Michael Farrell <micolous+git@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package au.id.micolous.metrodroid.transit.kmt;
+
+import android.app.Application;
+import android.os.Parcel;
+import android.support.annotation.Nullable;
+import net.kazzz.felica.lib.Util;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import au.id.micolous.farebot.R;
+import au.id.micolous.metrodroid.MetrodroidApplication;
+import au.id.micolous.metrodroid.card.felica.FelicaBlock;
+import au.id.micolous.metrodroid.transit.TransitCurrency;
+import au.id.micolous.metrodroid.transit.Trip;
+
+public class KMTTrip extends Trip {
+    public static final Creator<KMTTrip> CREATOR = new Creator<KMTTrip>() {
+        public KMTTrip createFromParcel(Parcel parcel) {
+            return new KMTTrip(parcel);
+        }
+
+        public KMTTrip[] newArray(int size) {
+            return new KMTTrip[size];
+        }
+    };
+    private final int mProcessType;
+    private final int mSequenceNumber;
+    private final Calendar mTimestamp;
+    private final int mTransactionAmount;
+    private final int mEndGateCode;
+
+    public KMTTrip(FelicaBlock block) {
+        byte[] data = block.getData();
+        mProcessType = data[12];
```

Comment:
`data[12] % 0xFF` will get rid of nasty surprises.

Reply:
& 0xff, not % 0xff